fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -416,35 +403,28 @@ def reduce(*_: Any) -> None:
 
                     # Make sure that this is not fired twice
                     self._grad_to_be_reduced[index] = False
-                    param.grad.mul_(self.world_size_scaling)
 
-                    if self.reduce_fp16:
-                        param.grad.data = param.grad.data.half()
+                    param.grad.mul_(self.world_size_scaling)
```

Comment:
What happened to the` if self.reduce_fp16`

Reply:
it did not give the correct results, I did not test it well enough. It can be re-implemented on top of this PR, I was genuinely trying to keep things "small"

Reply:
resolved in the newer version